common/scripted_effects/babylon_spontaneous_civil_war.txt

```diff
@@ -2,36 +2,224 @@
 	random_scope_state = {
 		limit = {
 			is_capital = no
+			is_incorporated = yes
 		}
-		save_scope_as = civil_war_center
+		set_variable = dynamic_country_cede
 	}
-	create_country = {
-		origin = root
-		state = scope:civil_war_center
-		on_created = {
-			set_variable = civil_war_country
+	every_scope_state = {
+		limit = {
+			is_capital = no
+			is_incorporated = yes
+			NOT = {
+				has_variable = dynamic_country_cede
+			}
+			any_neighbouring_state = {
+				is_incorporated = yes
+				has_variable = dynamic_country_cede
+			}
+		}
+		random = {
+			chance = $STREMGTH$
+			set_variable = dynamic_country_cede
 		}
 	}
-	random_country = {
+	every_scope_state = {
 		limit = {
-			has_variable = civil_war_country
+			is_capital = no
+			is_incorporated = yes
+			NOT = {
+				has_variable = dynamic_country_cede
+			}
+			any_neighbouring_state = {
+				is_incorporated = yes
+				has_variable = dynamic_country_cede
+			}
+		}
+		random = {
+			chance = $STREMGTH$
+			set_variable = dynamic_country_cede
 		}
-		save_scope_as = civil_war_country
 	}
 	every_scope_state = {
```

Comment:
Doing this twice changes the math,, the upper limit is now the  (1 - STRENGTH)^3 to not join province joins (which is means 87.5% of the states join the opposite side with 50 strength) so you should lower the strength (from 50) if this is supposed to be a equal war.

quick calculation is that the strength needs to be 16 to have an upper limit of 50%, it should be higher because realistically not all states are connected. But definitely not 50 xd

Reply:
The thing is, only provinces adjacent to a rebelling province gets the opportunity to join the rebellion..
I did it this way to make the rebels more cohesive in their territory.

Although I do admit the x3 is pulled from nowhere, in practice it gives a good amount of territory, which seems to hover around 1/3 of the united countrys GDP, typically.

Reply:
Ah I didn't see it starts from 1 province. We should note this, as this is highly unfavorable for smaller countries (where the amount off connected states approaches the total amount of states)